storage/cachestore/cachestore.go

```diff
@@ -44,21 +44,25 @@ func New(base blob.Store, maxBytes int) Store {
 	return Store{base: base, maxBytes: maxBytes}
 }
 
-// Keyspace implements a method of the [blob.Store] interface.
-// A successful result has concrete type [*KV] or [CAS], depending
-// whether the [blob.KV] returned by the base store implements [blob.CAS].
-// Each keyspace has its own separate cache.
-func (s Store) Keyspace(ctx context.Context, name string) (blob.KV, error) {
-	kv, err := s.base.Keyspace(ctx, name)
+// KV implements a method of the [blob.Store] interface.  A successful result
+// has concrete type [*KV].  Each keyspace has its own separate cache.
+func (s Store) KV(ctx context.Context, name string) (blob.KV, error) {
+	kv, err := s.base.KV(ctx, name)
 	if err != nil {
 		return nil, err
 	}
-	if cas, ok := kv.(blob.CAS); ok {
-		return NewCAS(cas, s.maxBytes), nil
-	}
 	return NewKV(kv, s.maxBytes), nil
 }
 
+// CAS implements a method of the [blob.Store] interface.
+func (s Store) CAS(ctx context.Context, name string) (blob.CAS, error) {
+	kv, err := s.KV(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	return blob.CASFromKV(kv), nil
```

Comment:
A thought: the error handling on getting a KV makes this boilerplate function a smidge long. If the signature of `CASFromKV` were instead `func CASFromKV(kv KV, err error) (CAS, error)`, you can benefit from the multi-value return corner case in the spec and make this function body `return blob.CASFromKV(s.KV(ctx, name))`. Same for other stores, the CAS method becomes a terse one-liner call to the common helper.

Reply:
Yes, now that we don't need CASKey to be able to transit an RPC boundary, we can remove the error. It was there mainly for the same reason that context is. (I'm inclined to leave the context in-place, since it is already everywhere in this API, but I agree the error could go away)

Reply:
Added `CASFromKVError` and propagated it where relevant.